tests/selfdestruct_test.go

```diff
@@ -0,0 +1,477 @@
+				},
+			},
+			effects: map[string]effectFunction{
+				"storage is not deleted":       contractStorageIs(big.NewInt(123)),
+				"code is not deleted":          contractCodeSizeIsNot(big.NewInt(0)),
+				"contract looses balance":      contractBalanceIs(big.NewInt(0)),
+				"beneficiary gains balance":    beneficiaryBalanceIs(someBalance),
+				"nested execution frame halts": executionHalted(),
+			},
+		},
+		"different tx/beneficiary is same account": {
+			transactions: []executeTxFunction[selfdestruct.SelfDestructFactory]{
+				func(factory *selfdestruct.SelfDestructFactory, opts *bind.TransactOpts, _ common.Address) (*types.Transaction, error) {
+					opts.Value = someBalance
+					return factory.Create(opts)
+				},
+				func(factory *selfdestruct.SelfDestructFactory, opts *bind.TransactOpts, _ common.Address) (*types.Transaction, error) {
+					return factory.DestroyWithoutBeneficiary(opts)
+				},
+			},
+			effects: map[string]effectFunction{
+				"storage is not deleted":       contractStorageIs(big.NewInt(123)),
+				"code is not deleted":          contractCodeSizeIsNot(big.NewInt(0)),
+				"balance is not burned":        contractBalanceIs(someBalance),
+				"nested execution frame halts": executionHalted(),
+			},
+		},
+		"same tx/beneficiary is other account": {
+			transactions: []executeTxFunction[selfdestruct.SelfDestructFactory]{
+				func(factory *selfdestruct.SelfDestructFactory, opts *bind.TransactOpts, otherAddress common.Address) (*types.Transaction, error) {
+					opts.Value = someBalance
+					return factory.CreateAndDestroy(opts, otherAddress)
+				},
+			},
+			effects: map[string]effectFunction{
+				"code is deleted":                contractCodeSizeIs(big.NewInt(0)),
+				"contract looses balance":        contractBalanceIs(big.NewInt(0)),
+				"beneficiary gains balance":      beneficiaryBalanceIs(someBalance),
+				"storage exists until end of tx": nestedContractValueAfterSelfDestructIs(big.NewInt(123)),
+				"nested execution frame halts":   executionHalted(),
+			},
+		},
+		"same tx/beneficiary is same account": {
+			transactions: []executeTxFunction[selfdestruct.SelfDestructFactory]{
+				func(factory *selfdestruct.SelfDestructFactory, opts *bind.TransactOpts, _ common.Address) (*types.Transaction, error) {
+					opts.Value = someBalance
+					return factory.CreateAndDestroyWithoutBeneficiary(opts)
+				},
+			},
+			effects: map[string]effectFunction{
+				"code is deleted":                contractCodeSizeIs(big.NewInt(0)),
+				"contract looses balance":        contractBalanceIs(big.NewInt(0)),
+				"storage exists until end of tx": nestedContractValueAfterSelfDestructIs(big.NewInt(123)),
+				"nested execution frame halts":   executionHalted(),
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			// generate a new beneficiary address for each test
+			otherAddress := common.BigToAddress(big.NewInt(int64(addressCount)))
+			addressCount++
```

Comment:
if this was not a global variable, the test could be executed in parallel

Reply:
I think that you are right.  The transaction order is only important for each test case. 
I will have a look.

Reply:
I tried to have unique beneficiary address to enable parallelism:
```
  selfdestruct_test.go:145: 
        	Error Trace:	/home/luis/code/Sonic/tests/selfdestruct_test.go:145
        	Error:      	Received unexpected error:
        	            	failed to deploy contract: replacement transaction underpriced
        	Test:       	TestSelfDestruct/constructor
WARN [11-05|09:02:02.710] Served eth_sendRawTransaction            conn=127.0.0.1:51156 reqid=1 duration="514.707µs" err="replacement transaction underpriced"
```

Unfortunately this fails to submit transactions to the network. I presume that there is an issue in the integration testing tool or in the summiting mechanism to the test validator. In theory this should be indeed parallel.